lms/models/group_info.py

```diff
@@ -40,9 +40,16 @@ class GroupInfo(BASE):
         nullable=False,
     )
 
+    #: The ApplicationInstance that this group belongs to foreign key
+    application_instance_id = sa.Column(
+        sa.Integer(),
+        sa.ForeignKey("application_instances.id", ondelete="cascade"),
+        nullable=True,
+    )
+
     #: The ApplicationInstance that this group belongs to.
     application_instance = sa.orm.relationship(
-        "ApplicationInstance", back_populates="group_infos"
+        "ApplicationInstance", back_populates="group_infos", foreign_keys=[consumer_key]
```

Comment:
This is the other side of the relationship, `foreign_keys` also needed here to know which FK to follow.

In this PR, as application_instance_id is null-able everything still uses `consumer_key`